addons/html_editor/static/tests/table/selection.test.js

```diff
@@ -1099,3 +1099,401 @@ describe("select columns on cross over", () => {
+                                <td><br></td>
+                            </tr>
+                        </tbody>
+                    </table>
+                `),
+            });
+            await testEditor({
+                contentBefore: unformat(`
+                    <table>
+                        <tbody>
+                            <tr>
+                                <td>
+                                    <p>abc</p>
+                                    <p>def</p>
+                                </td>
+                                <td><br></td>
+                            </tr>
+                            <tr>
+                                <td>abc[]</td>
+                                <td><br></td>
+                            </tr>
+                        </tbody>
+                    </table>
+                `),
+                stepFunction: async () => press("ArrowUp"),
+                contentAfter: unformat(`
+                    <table>
+                        <tbody>
+                            <tr>
+                                <td>
+                                    <p>abc</p>
+                                    <p>def[]</p>
+                                </td>
+                                <td><br></td>
+                            </tr>
+                            <tr>
+                                <td>abc</td>
+                                <td><br></td>
+                            </tr>
+                        </tbody>
+                    </table>
+                `),
+            });
+        });
+        test("should move cursor to the previous sibling of table", async () => {
+            await testEditor({
+                contentBefore: unformat(`
+                    <p>abcd</p>
+                    <table>
+                        <tbody>
+                            <tr>
+                                <td>[]<br></td>
+                                <td><br></td>
+                            </tr>
+                            <tr>
+                                <td><br></td>
+                                <td><br></td>
+                            </tr>
+                        </tbody>
+                    </table>
+                `),
+                stepFunction: async () => press("ArrowUp"),
+                contentAfter: unformat(`
+                    <p>abcd[]</p>
+                    <table>
+                        <tbody>
+                            <tr>
+                                <td><br></td>
+                                <td><br></td>
+                            </tr>
+                            <tr>
+                                <td><br></td>
+                                <td><br></td>
+                            </tr>
+                        </tbody>
+                    </table>
+                `),
+            });
+        });
+        test("should move cursor to the end cell of sibling table", async () => {
```

Comment:
Ok I guess this one requires a little bit more than checking for the sibiling of the closest(tr) because in that case it needs to check for a table sibling of the closest(table), but it's still not as complicated as the current proposition with the dompathgenerator etc.

(same for arrow down)

Reply:
Yes, that's correct. However, if the cursor is not at the first line of the current cell, pressing the `ArrowUp` key will cause the cursor to skip over any other elements in the current cell and directly move to the sibling table, ignoring the content within the current cell.

(vice-versa for arrow down).